blob-client/src/main/java/com/bazaarvoice/emodb/blob/client/BlobStoreClientFactory.java

```diff
@@ -1,50 +1,27 @@
 package com.bazaarvoice.emodb.blob.client;
 
 import com.bazaarvoice.emodb.common.jersey.dropwizard.JerseyEmoClient;
-import com.codahale.metrics.MetricRegistry;
 import com.google.common.base.Predicates;
 import com.google.common.base.Throwables;
 import com.google.common.collect.Iterables;
-import com.sun.jersey.api.client.Client;
-import com.sun.jersey.api.client.ClientHandlerException;
-import com.sun.jersey.api.client.UniformInterfaceException;
-import com.sun.jersey.client.apache4.ApacheHttpClient4;
-import com.sun.jersey.client.apache4.ApacheHttpClient4Handler;
-import com.sun.jersey.client.apache4.config.ApacheHttpClient4Config;
-import com.sun.jersey.client.apache4.config.DefaultApacheHttpClient4Config;
-import io.dropwizard.client.HttpClientBuilder;
-import io.dropwizard.client.HttpClientConfiguration;
-import io.dropwizard.jackson.Jackson;
-import io.dropwizard.jersey.jackson.JacksonMessageBodyProvider;
-import io.dropwizard.util.Duration;
-import org.apache.http.client.HttpClient;
+import org.glassfish.jersey.client.ClientProperties;
 
-import javax.validation.Validation;
-import javax.validation.ValidatorFactory;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.client.Client;
 import java.util.concurrent.ScheduledExecutorService;
 
 public class BlobStoreClientFactory extends AbstractBlobStoreClientFactory {
 
-    private static ValidatorFactory _validatorFactory = Validation.buildDefaultValidatorFactory();
-
-    public static BlobStoreClientFactory forCluster(String clusterName, MetricRegistry metricRegistry) {
-        HttpClientConfiguration httpClientConfiguration = new HttpClientConfiguration();
-        httpClientConfiguration.setKeepAlive(Duration.seconds(1));
-        return new BlobStoreClientFactory(clusterName, createDefaultJerseyClient(httpClientConfiguration, metricRegistry, getServiceName(clusterName)));
-    }
-
     /**
      * Connects to the Blob Store using the specified Jersey client.  If you're using Dropwizard, use this
      * constructor and pass the Dropwizard-constructed Jersey client.
      */
     public static BlobStoreClientFactory forClusterAndHttpClient(String clusterName, Client client) {
+        client.property(ClientProperties.SUPPRESS_HTTP_COMPLIANCE_VALIDATION, true);
```

Comment:
here and in others `*ClientFactory` classes:
it's not required, but let's think if it's possible to not mutate incoming parameter

Reply:
I think one option could be assign the client property, while creating the client with JerseyClientBuilder, like this.

`Client client = new JerseyClientBuilder(environment)
                    .using(configuration)
                    .using(environment)
                    .withProperty(ClientProperties.SUPPRESS_HTTP_COMPLIANCE_VALIDATION, true)
                    .build($name);
                    `